tpl/tplimpl/embedded/templates/opengraph.html

```diff
@@ -2,18 +2,26 @@
 <meta property="og:description" content="{{ with .Description }}{{ . }}{{ else }}{{if .IsPage}}{{ .Summary }}{{ else }}{{ with .Site.Params.description }}{{ . }}{{ end }}{{ end }}{{ end }}" />
 <meta property="og:type" content="{{ if .IsPage }}article{{ else }}website{{ end }}" />
 <meta property="og:url" content="{{ .Permalink }}" />
-{{- with .Params.images }}{{ range first 6 . }}
+{{ with .Params.images }}{{ range first 6 . -}}
 <meta property="og:image" content="{{ . | absURL }}" />
-{{ end }}{{ end }}
+{{ end }}{{ else -}}
+{{- $images := $.Resources.ByType "image" -}}
+{{- $featured := $images.GetMatch "*feature*" -}}
+{{- $featured := cond (ne $featured nil) $featured ($images.GetMatch "{*cover*,*thumbnail*}") -}}
```

Comment:
This will be slightly faster if you do:

```
{{ $if not $featured }}
{{- $featured = $images.GetMatch "{*cover*,*thumbnail*}" -}}
```

Reply:
@bep Did you mean the following change (or `$if` is a thing in Hugo)?

```diff
-{{- $featured := cond (ne $featured nil) $featured ($images.GetMatch "{*cover*,*thumbnail*}") -}}
+{{- if not $featured -}}
+{{- $featured = $images.GetMatch "{*cover*,*thumbnail*}" -}}
+{{- end -}}
```

Reply:
OK, first: I'm sorry that I let this hanging. We need to add this to Hugo.

What I meant above was:

```bash
{{ $if not $featured }}
{{- $featured = $images.GetMatch "{*cover*,*thumbnail*}" -}}
```